packages/remix-dev/compiler/dependencies.ts

```diff
@@ -1,21 +1,54 @@
+import * as fs from "fs";
 import * as path from "path";
-// @ts-expect-error
-import readPackageJson from "read-package-json-fast";
 
 import type { RemixConfig } from "../config";
 
 type PackageDependencies = { [packageName: string]: string };
 
-export async function getPackageDependencies(
+export function getPackageDependencies(
   packageJsonFile: string
-): Promise<PackageDependencies> {
-  return (await readPackageJson(packageJsonFile))?.dependencies || {};
+): PackageDependencies {
+  let pkg = JSON.parse(fs.readFileSync(packageJsonFile, "utf8"));
+  return pkg?.dependencies || {};
 }
 
-export function getAppDependencies(
-  config: RemixConfig
-): Promise<PackageDependencies> {
+export function getAppDependencies(config: RemixConfig): PackageDependencies {
   return getPackageDependencies(
     path.resolve(config.rootDirectory, "package.json")
   );
 }
+
+export function getDependenciesToBundle(...pkg: string[]): string[] {
+  let aggregatedDeps = new Set<string>(pkg);
+  let visitedPackages = new Set<string>();
+
+  pkg.forEach((p) => {
+    getPackageDependenciesRecursive(p, aggregatedDeps, visitedPackages);
+  });
+
+  return Array.from(aggregatedDeps);
+}
+
+function getPackageDependenciesRecursive(
+  pkg: string,
+  aggregatedDeps: Set<string>,
+  visitedPackages: Set<string>
+): void {
+  visitedPackages.add(pkg);
+
+  let pkgJson = require.resolve(`${pkg}/package.json`);
```

Comment:
Should probably use a `path.join` instead of a `/` here.

Reply:
Path.join will do a "\" on windows, and since this is actually require syntax, not a path, that would break.